src/rail/estimation/algos/deepdisc.py

```diff
@@ -35,104 +155,104 @@ class DeepDiscInformer(CatInformer):
+            self.set_data("input", training_data)
+            self.set_data("metadata", metadata)
             self.run()
             self.finalize()
-        return self.get_handle('model')
+        return self.get_handle("model")
 
     def finalize(self):
         pass
-    
+
     def run(self):
         """
         Train a inception NN on a fraction of the training data
         """
-        #train_data = self.get_data("input")
+        # train_data = self.get_data("input")
         metadata = self.get_data("metadata")
 
-        print('caching data')
+        print("Caching data")
         # create an iterator here
         itr = self.input_iterator("input")
         for start_idx, _, chunk in itr:
-            for idx, image in enumerate(chunk['images']):
+            for idx, image in enumerate(chunk["images"]):
                 this_img_metadata = metadata[start_idx + idx]
                 image_height = this_img_metadata["height"]
                 image_width = this_img_metadata["width"]
 
-                reformed_image = image.reshape(6, image_height, image_width).astype(np.float32)
+                reformed_image = image.reshape(6, image_height, image_width).astype(
+                    np.float32
+                )
 
-                filename = f'image_{start_idx + idx}.npy'
+                filename = f"image_{start_idx + idx}.npy"
                 file_path = os.path.join(self.temp_dir, filename)
                 np.save(file_path, reformed_image)
 
                 # we want the dictionary associated with this particular image.
-                metadata[start_idx + idx]['filename'] = file_path
+                metadata[start_idx + idx]["filename"] = file_path
 
-        cfgfile = self.config.cfgfile
-        batch_size = self.config.batch_size
-        numclasses = self.config.numclasses
-        epochs = self.config.epochs
-        output_dir = self.config.output_dir
-        output_name = self.config.output_name
+        self.metadata = metadata
 
-        val_per = 1000 # Should be included as an input config parameter.
+        num_gpus = self.config.num_gpus
+        num_machines = 1
+        machine_rank = self.config.machine_rank
 
-        cfg = get_lazy_config(cfgfile, batch_size, numclasses)
-        cfg_loader = get_loader_config(output_dir, batch_size, epochs)
-
-        model = return_lazy_model(cfg)
-        cfg.optimizer.params.model = model
-        cfg.optimizer.lr = 0.001
-        # optimizer = return_optimizer(cfg)
-        optimizer = solver.build_optimizer(cfg_loader, model)
-      
-        #When using the single test dictionary, add this code and replace "mapper" below
-        mapper = RedshiftDictMapper(
-            DC2ImageReader(), lambda dataset_dict: dataset_dict["filename"]
-        ).map_data
-
-        # mapper = RedshiftFlatDictMapper().map_data
-
-        loader = d2data.build_detection_train_loader(
-            metadata, mapper=mapper, total_batch_size=batch_size
+        port = (
```

Comment:
This looks like magic. It may be nice to just have a distinct function called `get_dist_url` that will return the dist_url string and encapsulate all the math and concatenation. Doesn't have to happen in this PR though.